migrate_test.go

```diff
@@ -233,33 +233,33 @@ func TestMigrate(t *testing.T) {
 		expectSeq     migrationSequence
 	}{
 		// migrate all the way Up in single steps
-		{version: 0, expectErr: os.ErrNotExist},
-		{version: 1, expectErr: nil, expectVersion: 1, expectSeq: seq.add(M(1))},
-		{version: 2, expectErr: os.ErrNotExist},
-		{version: 3, expectErr: nil, expectVersion: 3, expectSeq: seq.add(M(3))},
-		{version: 4, expectErr: nil, expectVersion: 4, expectSeq: seq.add(M(4))},
+		{version: 0, expectErr: os.ErrNotExist, expectSeq: seq.add()},
+		{version: 1, expectErr: nil, expectVersion: 1, expectSeq: seq.add(MR("CREATE 1"))},
+		{version: 2, expectErr: os.ErrNotExist, expectSeq: seq.add()},
+		{version: 3, expectErr: nil, expectVersion: 3, expectSeq: seq.add(MR("CREATE 3"))},
+		{version: 4, expectErr: nil, expectVersion: 4, expectSeq: seq.add(MR("CREATE 4"))},
 		{version: 5, expectErr: nil, expectVersion: 5, expectSeq: seq.add()}, // 5 has no up migration
-		{version: 6, expectErr: os.ErrNotExist},
-		{version: 7, expectErr: nil, expectVersion: 7, expectSeq: seq.add(M(7))},
-		{version: 8, expectErr: os.ErrNotExist},
+		{version: 6, expectErr: os.ErrNotExist, expectSeq: seq.add()},
+		{version: 7, expectErr: nil, expectVersion: 7, expectSeq: seq.add(MR("CREATE 7"))},
+		{version: 8, expectErr: os.ErrNotExist, expectSeq: seq.add()},
 
 		// migrate all the way Down in single steps
-		{version: 6, expectErr: os.ErrNotExist},
-		{version: 5, expectErr: nil, expectVersion: 5, expectSeq: seq.add(M(7, 5))},
-		{version: 4, expectErr: nil, expectVersion: 4, expectSeq: seq.add(M(5, 4))},
-		{version: 3, expectErr: nil, expectVersion: 3, expectSeq: seq.add(M(4, 3))},
-		{version: 2, expectErr: os.ErrNotExist},
+		{version: 6, expectErr: os.ErrNotExist, expectSeq: seq.add()},
+		{version: 5, expectErr: nil, expectVersion: 5, expectSeq: seq.add(MR("DROP 7"))},
+		{version: 4, expectErr: nil, expectVersion: 4, expectSeq: seq.add(MR("DROP 5"))},
+		{version: 3, expectErr: nil, expectVersion: 3, expectSeq: seq.add(MR("DROP 4"))},
+		{version: 2, expectErr: os.ErrNotExist, expectSeq: seq.add()},
 		{version: 1, expectErr: nil, expectVersion: 1, expectSeq: seq.add()}, // 3 has no down migration
-		{version: 0, expectErr: os.ErrNotExist},
+		{version: 0, expectErr: os.ErrNotExist, expectSeq: seq.add()},
 
 		// migrate all the way Up in one step
-		{version: 7, expectErr: nil, expectVersion: 7, expectSeq: seq.add(M(3), M(4), M(7))},
+		{version: 7, expectErr: nil, expectVersion: 7, expectSeq: seq.add(MR("CREATE 3"), MR("CREATE 4"), MR("CREATE 7"))},
 
 		// migrate all the way Down in one step
-		{version: 1, expectErr: nil, expectVersion: 1, expectSeq: seq.add(M(7, 5), M(5, 4), M(4, 3), M(3, 1))},
```

Comment:
`M(3, 1)` now goes away since as far as I can tell it never had an effect anyway.